### ISSUE TITLE
Port FtpWebRequest/Response

### ISSUE THREAD

**Issue by karelz** [dotnet/corefx#11883]
Should be implemented on top of TCP (start with Desktop .NET code which is on top of Sockets).
Note: There are better OSS APIs, but with very different API surface.

**Comment by ericeil** [comment]
```csharp
    public enum FtpStatusCode {
        AccountNeeded = 532,
        ActionAbortedLocalProcessingError = 451,
        ActionAbortedUnknownPageType = 551,
        ActionNotTakenFilenameNotAllowed = 553,
        ActionNotTakenFileUnavailable = 550,
        ActionNotTakenFileUnavailableOrBusy = 450,
        ActionNotTakenInsufficientSpace = 452,
        ArgumentSyntaxError = 501,
        BadCommandSequence = 503,
        CantOpenData = 425,
        ClosingControl = 221,
        ClosingData = 226,
        CommandExtraneous = 202,
        CommandNotImplemented = 502,
        CommandOK = 200,
        CommandSyntaxError = 500,
        ConnectionClosed = 426,
        DataAlreadyOpen = 125,
        DirectoryStatus = 212,
        EnteringPassive = 227,
        FileActionAborted = 552,
        FileActionOK = 250,
        FileCommandPending = 350,
        FileStatus = 213,
        LoggedInProceed = 230,
        NeedLoginAccount = 332,
        NotLoggedIn = 530,
        OpeningData = 150,
        PathnameCreated = 257,
        RestartMarker = 110,
        SendPasswordCommand = 331,
        SendUserCommand = 220,
        ServerWantsSecureSession = 234,
        ServiceNotAvailable = 421,
        ServiceTemporarilyNotAvailable = 120,
        SystemType = 215,
        Undefined = 0,
    }
    public sealed class FtpWebRequest : WebRequest {
        public X509CertificateCollection ClientCertificates { get; set; }
        public override string ConnectionGroupName { get; set; }
        public override long ContentLength { get; set; }
        public long ContentOffset { get; set; }
        public override string ContentType { get; set; }
        public override ICredentials Credentials { get; set; }
        public static new RequestCachePolicy DefaultCachePolicy { get; set; }
        public bool EnableSsl { get; set; }
        public override WebHeaderCollection Headers { get; set; }
        public bool KeepAlive { get; set; }
        public override string Method { get; set; }
        public override bool PreAuthenticate { get; set; }
        public override IWebProxy Proxy { get; set; }
        public int ReadWriteTimeout { get; set; }
        public string RenameTo { get; set; }
        public override Uri RequestUri { get; }
        public ServicePoint ServicePoint { get; }
        public override int Timeout { get; set; }
        public bool UseBinary { get; set; }
        public override bool UseDefaultCredentials { get; set; }
        public bool UsePassive { get; set; }
        public override void Abort();
        public override IAsyncResult BeginGetRequestStream(AsyncCallback callback, object state);
        public override IAsyncResult BeginGetResponse(AsyncCallback callback, object state);
        public override Stream EndGetRequestStream(IAsyncResult asyncResult);
        public override WebResponse EndGetResponse(IAsyncResult asyncResult);
        public override Stream GetRequestStream();
        public override WebResponse GetResponse();
    }
    public class FtpWebResponse : WebResponse, IDisposable {
        public string BannerMessage { get; }
        public override long ContentLength { get; }
        public string ExitMessage { get; }
        public override WebHeaderCollection Headers { get; }
        public DateTime LastModified { get; }
        public override Uri ResponseUri { get; }
        public FtpStatusCode StatusCode { get; }
        public string StatusDescription { get; }
        public override bool SupportsHeaders { get; }
        public string WelcomeMessage { get; }
        public override void Close();
        public override Stream GetResponseStream();
    }
```

**Comment by geoffkizer** [comment]
Implemented in #12594
